src/org/labkey/remoteapi/issues/IssueModel.java

```diff
@@ -0,0 +1,213 @@
+    {
+        _properties.put(ISSUE_ID, issueId);
+        return this;
+    }
+
+    public String getTitle()
+    {
+        return (String) _properties.get(TITLE);
+    }
+
+    public IssueModel setTitle(String title)
+    {
+        _properties.put(TITLE, title);
+        return this;
+    }
+
+    // per https://www.labkey.org/Documentation/wiki-page.view?name=sampleJSscripts#issues, for a single notify user
+    // you can supply an email address as a string; for multiple notifies, userIDs are necessary
+    public String getNotify()
+    {
+        return (String) _properties.get("notifyList");
+    }
+
+    /**
+     * when notify is just 1 user, you can provide their email
+     * @param notify
+     * @return
+     */
+    public IssueModel setNotify(String notify)
+    {
+        _properties.put(NOTIFY_LIST, notify);
+        return this;
+    }
+
+    public Integer getAssignedTo()
+    {
+        return (Integer)_properties.get(ASSIGNED_TO);
+    }
+
+    public IssueModel setAssignedTo(Long assignedTo)
+    {
+        _properties.put(ASSIGNED_TO, assignedTo);
+        return this;
+    }
+
+    public String getIssueDefName()
+    {
+        return (String) _properties.get(ISSUE_DEF_NAME);
+    }
+
+    public IssueModel setIssueDefName(String issueDefName)
+    {
+        _properties.put(ISSUE_DEF_NAME, issueDefName);
+        return this;
+    }
+
+    public String getType()
+    {
+        return _properties.get(TYPE).toString();
+    }
+
+    public IssueModel setType(String type)
+    {
+        return setProp(TYPE, type);
+    }
+
+    public Long getPriority()
+    {
+        return (Long) _properties.get(PRIORITY);
+    }
+
+    public IssueModel setPriority(Long priority)
+    {
+        return setProp(PRIORITY, priority);
+    }
+
+    public String getComment()
+    {
+        return _properties.get(COMMENT).toString();
+    }
+
+    public IssueModel setComment(String comment)
+    {
+        return setProp(COMMENT, comment);
+    }
+
+    public String getResolution()
+    {
+        return (String) _properties.get(RESOLUTION);
+    }
+
+    public IssueModel setResolution(String resolution)
+    {
+        return setProp(RESOLUTION, resolution);
+    }
+
+    public IssueModel setProp(String propName, Object value)
+    {
+        _properties.put(propName, value);
+        return this;
+    }
+
+    public IssueModel addAttachment(File attachmentFile)
+    {
+        _attachments.add(attachmentFile);
+        return this;
+    }
+
+    public List<File> getAttachments()
+    {
+        return _attachments;
+    }
+
+    public enum IssueAction
+    {
+        INSERT("insert"),
+        UPDATE("update"),
+        RESOLVE("resolve"),
+        CLOSE("close"),
+        REOPEN("reopen");
+
+        IssueAction(String value)
+        {
+            _value = value;
+        }
+        public String getValue()
+        {
+            return _value;
+        }
+        private final String _value;
```

Comment:
What's the value of exposing a `getValue` method, couldn't the `enum` just be the action literals? `Enums` also override `toString` so they can be used directly in places where a `String` is expected.

Reply:
I put the getter there not thinking that .toString() would be good form; if it's preferable I don't have opinions either way